pkg/agent/openflow/pipeline.go

```diff
@@ -922,6 +923,9 @@ func (c *client) l2ForwardCalcFlow(dstMAC net.HardwareAddr, ofPort uint32, skipI
 		// Go to ingress NetworkPolicy tables for traffic to local Pods.
 		nextTable = c.ingressEntryTable
 	}
+	if skipIngressRules && c.enableNetworkPolicyStats {
+		nextTable = IngressMetricTable
```

Comment:
The metric flow is currently installed unconditionally and the request packets go through the metric table anyway so it would be better to keep it consistent when fixing the issue. Maybe just make IngressMetricTable as the next table of l2ForwardingCalcTable.
When necessary, we can change to initialize metric tables and make all traffic go thought them based on the feature gate together.